server/container/guice/guice-common/src/test/java/org/apache/james/MailsShouldBeWellReceived.java

```diff
@@ -55,6 +71,67 @@
         .await();
 
     ConditionFactory CALMLY_AWAIT_FIVE_MINUTE = CALMLY_AWAIT.timeout(Duration.FIVE_MINUTES);
+    String SENDER = "bob@apache.org";
+    String UNICODE_BODY = "A 8bit encoded body with €uro symbol.";
+
+
+    static String readFirstMessageJavax(int imapPort) throws MessagingException, IOException {
+        Session imapSession = Session.getDefaultInstance(new Properties());
+        Store store = imapSession.getStore("imap");
+        store.connect("localhost", imapPort, JAMES_USER, PASSWORD);
+        Folder inbox = store.getFolder(IMAPMessageReader.INBOX);
+        inbox.open(Folder.READ_ONLY);
+
+        CALMLY_AWAIT.untilAsserted(() ->
+            assertThat(searchForAll(inbox))
+                .hasSize(1));
+
+        return IOUtils.toString(
+            searchForAll(inbox)[0].getInputStream(),
+            StandardCharsets.UTF_8);
```

Comment:
the trick is here: you store raw data in UTF-8, and ask to retrieve it as UTF-8. Hopefully it works :)

Reply:
Okay, should use the standard way to parse IMAP message